markymark/src/main/java/com/m2mobi/markymark/annotator/MarkyMarkAnnotator.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2022 M2mobi
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the “Software”), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+package com.m2mobi.markymark.annotator
+
+import androidx.compose.ui.text.AnnotatedString
+import com.m2mobi.markymark.model.AnnotatedStableNode
+import com.m2mobi.markymark.theme.AnnotatedStyles
+
+/**
+ * The annotator is responsible for rendering [AnnotatedStableNode]s. See [DefaultMarkyMarkAnnotator] for the default
+ * implementation.
+ */
+interface MarkyMarkAnnotator {
+
+    /**
+     * Create [AnnotatedString] for the [nodes].
+     */
+    fun annotate(
```

Comment:
Add `@Stable` here since the function is called from a `@Composable`.

Think the interface can have it too?

Reply:
The interface has it but yeah good one. Will add it to the function too

Reply:
Added it to the implementation class + function too